src/VariationPriceHandler.php

```diff
@@ -0,0 +1,211 @@
+    $this->logger = $logger_factory->get('system');
+    $this->entityTypeManager = $entity_type_manager;
+    $this->entityManager = $entity_manager;
+    $this->attributeFieldManager = $attribute_field_manager;
+  }
+
+  /**
+   * Retrieves the product variations.
+   *
+   * @param array $types
+   *   The product variation types.
+   * @param string $attribute
+   *   The attribute machine name.
+   * @param int $attributeValue
+   *   The Attribue value id.
+   *
+   * @return array
+   *   Array of variations id.
+   */
+  public function getProductVariation(array $types, $attribute, $attributeValue) {
+    // The attribute field name of variation type will always be in the
+    // format of `attrubute_attribute_machine_name`.
+    $attributefieldName = 'attribute_' . $attribute;
+
+    // Get all variations associated with an particular attribute value.
+    $query = \Drupal::entityQuery('commerce_product_variation')
+      ->condition('type', $types, 'IN')
+      ->condition($attributefieldName, $attributeValue);
+    $variations = $query->execute();
+    return $variations;
+  }
+
+  /**
+   * Recalculates the product variation price based on the attribute price.
+   *
+   * @param Drupal\commerce_attributes\Entity\ProductAttributeValue $attributeValue
+   *   The ProductAttributeValue entity.
+   */
+  public function updateProductVariationsPrice(ProductAttributeValue $attributeValue) {
+    $attribute = $attributeValue->getAttribute();
+
+    // Get all variation type associated with the attribute
+    // which is being updated.
+    $variationTypes = $this->getAllVariationTypes($attribute);
+    if (empty($variationTypes)) {
+      return FALSE;
+    }
+
+    // Get all product variations associated with the attribute value which is
+    // being updated.
+    $variations = $this->getProductVariation($variationTypes, $attribute->id(), $attributeValue->id());
+    foreach ($variations as $variation_id) {
+      $product_variation = $this->entityManager->getStorage('commerce_product_variation')->load($variation_id);
+      $product_id = $product_variation->get('product_id')->getValue()['0']['target_id'];
+      $product = $this->entityManager->getStorage('commerce_product')->load($product_id);
+      $productBaseCost = $product->get('field_price')->getValue()['0']['number'];
+
+      // Get the total price of all attributes associated with the product
+      // variation type.
+      $attributePrice = $this->getAttributePriceSum($product_variation->bundle(), $product_variation);
+      $newPrice = $productBaseCost + $attributePrice;
+      $currency = $product->get('field_price')->getValue()['0']['currency_code'];
+      $newPrice = new Price((string) $newPrice, $currency);
+      $product_variation->setPrice($newPrice);
+      $product_variation->save();
```

Comment:
**CHANGE REQUEST**

First, rename the field to just `price`.
Price objects are able to do maths. The attribute price sum should be a Price object. Then you get the price like:

`$base_price = $product->get('price')->toPrice();` or if you get an error ``$base_price = $product->get('price')->first()->toPrice();`

Then add them using the Price API:
`$new_price = $base_price->add($attributes_price);`

Reply:
Done